Source/JavaScriptCore/assembler/MacroAssemblerARMv7.h

```diff
@@ -947,12 +999,12 @@ class MacroAssemblerARMv7 : public AbstractMacroAssembler<Assembler> {
                     cachedDataTempRegister().invalidate();
                 m_assembler.ldrd(dest1, dest2, address.base, address.u.offset, /* index: */ true, /* wback: */ false);
             } else if (address.base == dest1) {
-                ArmAddress highAddress(address.base, address.u.offset + 4);
+                ArmAddress highAddress(address.base, address.u.offset + TagOffset);
```

Comment:
This change is semantically not correct. Load pair should always load the 32-bit word at the numerically higher address into `dest2`. (It is another matter that  we don't have JIT on big-endian CPUs where `TagOffset` and `PayloadOffset` would be swapped and this would cause issue.)

Same 3 more times below.